### PR TITLE
[enhance](nereids) date_add, date_sub, date_diff, date_floor, date_ceil function implement Monotonic

### PR DESCRIPTION
**Pull request by feiniaofeiafei**
### What problem does this PR solve?

date_add, date_sub, date_diff, date_floor, date_ceil function implement Monotonic, so we can do prune range partition for this functions, for example `date_add(dt, 1) = '2024-01-01'`

### Check List (For Author)

- Test 
    - [x] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [x] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by feiniaofeiafei** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/DayCeil.java [hunk 1]
```diff
@@ -106,4 +107,19 @@ public List<FunctionSignature> getSignatures() {
...
+        switch (arity()) {
+            case 1:
+                return new DayCeil(literal);
+            case 2:
+                return new DayCeil(literal, child(1));
+            case 3:
+                return new DayCeil(literal, child(1), child(2));
+            default:
+                // should not reach
+                return null;
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/DayCeil.java:122]: 1. Do LOG
2. throw exception or check null in caller

**Review by morrySnow — DISMISSED** [review]

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run p0

**Review by morrySnow — APPROVED** [review]

**Review by 924060929 — APPROVED** [review]

File: regression-test/suites/nereids_rules_p0/partition_prune/test_convert_tz.groovy [hunk 2]
```diff
@@ -93,7 +93,7 @@ suite("test_convert_tz") {
         }
         explain {
             sql "SELECT * FROM test_convert_tz WHERE not convert_tz(timestamp, 'Asia/Shanghai', 'Europe/Paris') <= '2021-01-01';";
-            contains("partitions=2/3 (p2,p3)")
+            contains("partitions=3/3 (p1,p2,p3)")
```
Comment (924060929) [inline regression-test/suites/nereids_rules_p0/partition_prune/test_convert_tz.groovy:96]: this test has rollback, we should improve it later

**Review by 924060929 — APPROVED** [review]

**Pull request merged**